sig/key.go

```diff
@@ -0,0 +1,56 @@
+package sig
+
+import (
+	"crypto/cipher"
+	"github.com/dedis/crypto/abstract"
+	"hash"
+)
+
+// PublicKey is the abstract interface for a public signing key
+// that can be used for signature verification.
+// This interface may be used with a variety of signing schemes,
+// but it assumes that the signing scheme produces and consumes
+// fixed-size signatures.
```

Comment:
Why must we assume that it produces and consume fixed length signature ? I see below you hash all the content using your writer, so the message signature is fixed size regarding the hash. 
Is it something 99% of signing scheme do ? I assume it has some reasoning regarding side-channel attacks ?